examples/cfd/01_convection.ipynb

```diff
@@ -120,14 +120,15 @@
     "    un = u.copy()\n",
     "    \n",
     "    # Update the new result with a 3-point stencil\n",
-    "    u[1:, 1:] = (un[1:, 1:] - (c * dt / dx * (un[1:, 1:] - un[1:, :-1])) -\n",
-    "                              (c * dt / dy * (un[1:, 1:] - un[:-1, 1:])))\n",
-    "\n",
-    "    # Apply boundary conditions\n",
-    "    u[0, :] = 1.\n",
-    "    u[-1, :] = 1.\n",
-    "    u[:, 0] = 1.\n",
-    "    u[:, -1] = 1."
+    "    u[1:, 1:] = (un[1:, 1:] - (c * dt / dy * (un[1:, 1:] - un[1:, :-1])) -\n",
+    "                              (c * dt / dx * (un[1:, 1:] - un[:-1, 1:])))\n",
+    "\n",
+    "    # Apply boundary conditions. \n",
+    "    # Note: -1 here is the last index in the array, not the one at x=-1 or y=-1.\n",
```

Comment:
I'd move this note to below the expressions and modify it slightly to something along the lines of:
"Note that in the above expressions the `NumPy` index `-1` corresponds to the final point of the array along the indexed dimension, i.e. here `u[-1, :]` is equivalent to `u[80, :]`."